CMSgov/design-system
# [WNMGDS-3027] Add Web Component Sections to doc site pages.

## Summary

This pull request is part of a series aimed at adding missing "Web Component" sections to component pages on the documentation site. By including these sections, we ensure developers have quick access to code examples and guidance for the web component version of each component.
- Added "Web Component" sections to the following component pages: Autocomplete, Checkbox, Date Field, Drawer, and Filter Chip.
- Renamed `ReactDocLink` and `ReactDocLinks` components to `StorybookDocLink` and `StorybookDocLinks`, respectively.  
  - Added logic to handle both React and Web Component cases when there are multiple links, such as links to Choice and ChoiceList.
- Corrected a broken link on the Drawer component page.

## How to test
Verify the Storybook links on the following pages to ensure they direct to the correct documentation for both React and Web Component versions:

### Pages with Added Web Component Sections
- [Autocomplete](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/autocomplete/?theme=core#code)
- [Checkbox](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/checkbox/?theme=core#code)
- [Date Field](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/date-field/single-input-date-field/?theme=core#code)
- [Drawer](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/drawer/?theme=core#code)
- [Filter Chip](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/filter-chip/?theme=core#code)
- [Tooltip](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/tooltip/?theme=core#code)
- [Tabs](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/tabs/?theme=core#code)

### Additional Pages with Updated Link Components
Since changes were made to the link components, please also verify the Storybook links on these pages to ensure consistent functionality:
- [Radio](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/radio/?theme=core#code)
- [Step List](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/step-list/?theme=core#code)
- [Table](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/table/?theme=core#code)
- [Vertical Navigation](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-3027/add-wc-sections-to-component-pages/components/vertical-navigation/?theme=core#code)

Jira story: [WNMGDS-3027](https://jira.cms.gov/browse/WNMGDS-3027)

## Checklist
